src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+) extends Metric {
+  private def divide(a: Double, b: Double): Double = {
+    if (b == 0) 0
+    else a / b
+  }
+  def finalizeMetric(): this.type = {
+    this.frac_phased                                   = this.divide(this.num_phased, this.num_called)
+    this.frac_phased_with_truth_phased                 = this.divide(this.num_phased_with_truth_phased, this.num_called_with_truth_phased)
+    this.frac_truth_phased_in_called_block             = this.divide(this.num_truth_phased_in_called_block, this.num_truth_phased)
+    this.frac_phased_with_truth_phased_in_called_block = this.divide(this.num_both_phased_in_called_block, this.num_truth_phased_in_called_block)
+    this
+  }
+}
+
+/** The cigar Elements */
+private object CigarType extends Enumeration {
+  type CigarType = Value
+  val Match,      // 0
+  Mismatch,       // 1
+  Truth,          // 2
+  Call,           // 3
+  TruthEnd,       // 4
+  CallEnd,        // 5
+  BothEnd = Value // 6
+}
+
+private[vcf] object Cigar {
+  import CigarType._
+
+  type Cigar = Seq[CigarType]
+
+  private type VCtx = VariantContext
+
+  /** Creates a Cigar from an iterator over variant contexts from a truth and call sample.
+    *
+    * If `skipMismatchingAlleles`, then it skips sites where both truth and call have a variant call, but the alleles
+    * disagree between the two.
+    *
+    * Examples of short and long errors:
+    *   CALL :  MMMMXXXMMM   MMMM|FFFFFFF|MMMMMFMMMM|F|MMFFFMMM
+    *   TRUTH:  MMMM   MMMMMMMMMM-MMMMMMM|MMMMMMMMMM|F|MMMMMMMM
+    *             correct       long        short     long
+    *   CIGAR:  00003330002220000511111116000001000060600111000
+    * shows one known block split
+    */
+  def toCigar(pairedIterator: JointVariantContextIterator,
+              truthPhaseBlockDetector: OverlapDetectorByContig,
+              callPhaseBlockDetector: OverlapDetectorByContig,
+              metric: AssessPhasingMetric,
+              skipMismatchingAlleles: Boolean): Cigar = {
+    val cigar: Cigar = pairedIterator
+      .filter { // ensure the alleles are the same when both truth and call are called
+        case Seq(Some(t: VCtx), Some(c: VCtx)) =>
+          import scala.collection.JavaConversions.iterableAsScalaIterable
+          !skipMismatchingAlleles || t.getAlleles.toSet == c.getAlleles.toSet
+        case _                                 => true
+      }
+      .flatMap { case Seq(t, c) =>
+        val ctxs                 = Seq(t, c).flatten // NB: can be 1 or 2 contexts here
+        val inTruthPhaseBlock    = ctxs.exists(truthPhaseBlockDetector.overlapsAny)
+        val inCallPhaseBlock     = ctxs.exists(callPhaseBlockDetector.overlapsAny)
+        val isTruthVariant       = t.isDefined
+        val isCallVariant        = c.isDefined
+        val truthPhasedVariant   = t.filter { ctx => PhasingUtils.getPhasingSetId(ctx) > 0 }
+        val callPhasedVariant    = c.filter { ctx => PhasingUtils.getPhasingSetId(ctx) > 0 }
```

Comment:
My brain keeps tripping up on these names.  I think it would be easier to read if you used `called` instead of `call` in places like this where your usage is as an adjective on a thing (variant, block).